core/services/eth/log_broadcaster.go

```diff
@@ -379,3 +384,22 @@ func (l *decodingLogListener) HandleLog(log interface{}, err error) {
 
 	l.LogListener.HandleLog(decodedLog, nil)
 }
+
+func appendLogChannel(ch1, ch2 chan eth.Log) chan eth.Log {
+	chCombined := make(chan eth.Log)
+
+	go func() {
```

Comment:
I think I would add a `defer close(chCombined)` here, rather than calling `close(chRawLogs)` above.  Did you try that out but not get anywhere with it?